python/cudf/cudf/core/frame.py

```diff
@@ -580,30 +454,19 @@ def equals(self, other, **kwargs):
         >>> df.equals(different_column_type)
         True
         """
-        if self is other:
-            return True
-
-        check_types = kwargs.get("check_types", True)
-
-        if check_types:
-            if type(self) is not type(other):
-                return False
-
-        if other is None or len(self) != len(other):
-            return False
-
-        # check data:
-        for self_col, other_col in zip(
-            self._data.values(), other._data.values()
+        if (
```

Comment:
I actually just removed that check because it's purely an optimization for an edge case that seems very unlikely. Why would someone bother to check `df.equals(df)`? And why do this check just to optimize for that case? If someone's code is being slowed by that, they can just avoid calling `equals`, right?

Reply:
One most used scenario is to `reindex`, users will have control over `dask` or `cudf` code that does perform a `Frame.equals(Frame)` check to perform a `reindex`. In which case keeping `self is other` will serve as a fast-path but removing it will have to let it go through all other checks.